### PR TITLE
tables: update Windows codename column to be that of name

### PR DESCRIPTION
**Pull request by muffins**
Summary:
This addresses Issue #4080 by removing the `code_name` column on Windows whole-sale, and replacing it's contents with that of the `name` column. See the discussion in #4080 for additional details. 

Test Plan:

Ran on Windows 10:
```
PS C:\Users\Nicholas\work\repos\osquery\build\windows10> .\osquery\RelWithDebInfo\osqueryd.exe -S                                                                 Using a [1mvirtual database[0m. Need help, type '.help'
osquery> select * from os_version;
+--------------------------+------------+-------+-------+-------+-------+----------+---------------+--------------------------+---------------------------+
| name                     | version    | major | minor | patch | build | platform | platform_like | codename                 | install_date              |
+--------------------------+------------+-------+-------+-------+-------+----------+---------------+--------------------------+---------------------------+
| Microsoft Windows 10 Pro | 10.0.18362 | 10    | 0     |       | 18362 | windows  | windows       | Microsoft Windows 10 Pro | 20190522202404.000000-420 |
+--------------------------+------------+-------+-------+-------+-------+----------+---------------+--------------------------+---------------------------+
osquery>
```
Ran on Windows 7:
```
PS C:\Users\RE\Desktop> .\osqueryd.exe -S
Using a ←[1mvirtual database←[0m. Need help, type '.help'
osquery> select * from os_version;
+---------------------------------+----------+-------+-------+-------+-------+----------+---------------+---------------
------------------+---------------------------+
| name                            | version  | major | minor | patch | build | platform | platform_like | codename
                  | install_date              |
+---------------------------------+----------+-------+-------+-------+-------+----------+---------------+---------------
------------------+---------------------------+
| Microsoft Windows 7 Enterprise  | 6.1.7601 | 6     | 1     |       | 7601  | windows  | windows       | Microsoft Wind
ows 7 Enterprise  | 20150317162012.000000-360 |
+---------------------------------+----------+-------+-------+-------+-------+----------+---------------+---------------
------------------+---------------------------+
osquery>
```

### CODE REVIEW

**Comment by theopolis** [comment]
I think you got bit by a flaky test:
```
11: Test command: /__w/1/b/build/osquery/logger/tests/osquery_logger_tests-test
11: Test timeout computed to be: 10000000
11: Running main() from /__w/1/b/build/libs/src/patched-source/googletest/src/googletest/src/gtest_main.cc
11: [==========] Running 12 tests from 1 test case.
11: [----------] Global test environment set-up.
11: [----------] 12 tests from LoggerTests
11: [ RUN      ] LoggerTests.test_plugin
11: [       OK ] LoggerTests.test_plugin (1 ms)
11: [ RUN      ] LoggerTests.test_logger_init
11: W1015 04:47:52.000437 57552 logger.cpp:152] Logger test is generating a warning status (1)
11: [       OK ] LoggerTests.test_logger_init (0 ms)
11: [ RUN      ] LoggerTests.test_log_string
11: [       OK ] LoggerTests.test_log_string (0 ms)
11: [ RUN      ] LoggerTests.test_logger_log_status
11: W1015 04:47:52.000552 57552 logger.cpp:186] Logger test is generating a warning status (2)
11: /__w/1/s/osquery/logger/tests/logger.cpp:194: Failure
11: Expected: (now) >= (LoggerTests::last_status.time), actual: 1571114871 vs 1571114872
11: [  FAILED  ] LoggerTests.test_logger_log_status (0 ms)
```

**Comment by muffins** [comment]
@keeleysam @FritzX6 @directionless y'all mind taking a look considering this is relevant to the issue discussing this problem? Curious of folks' thoughts on removing the list whole sale.

**Comment by keeleysam** [comment]
This looks like a completely reasonable change to me, and good to not maintain that list in the source code any longer.

File: osquery/tables/system/windows/os_version.cpp [hunk 1]
```diff
@@ -142,7 +31,10 @@ QueryData genOSVersion(QueryContext& context) {
   }
 
   wmiResults[0].GetString("InstallDate", r["install_date"]);
-  wmiResults[0].GetString("Caption", r["name"]);
+  std::string osName{""};
```
Comment (theopolis) [inline osquery/tables/system/windows/os_version.cpp]: I think the `{""}` is superfluous.

**Comment by theopolis** [comment]
@muffins, one nitpick, but do you mind rebasing onto master too?

**Comment by directionless** [comment]
I agree maintaining the mapping here seems hard. I don't know what the underlying intent is, but I'm generally onboard here. I also observe that removing the leading `Microsoft Windows` gets you pretty close to the codenames. With some weirdness around language specific editions. I'm curious if @FritzX6 weighs in.

```
Microsoft Windows 7 Ultimate ,Ultimate
Microsoft Windows 10 Pro for Workstations,
Microsoft Windows 10 Enterprise Insider Preview,Windows 10 Enterprise
Microsoft Windows 10 Education,
Microsoft Windows 10 Home,Windows 10 Home
Microsoft Windows 10 企业版 2016 长期服务版,
Microsoft Windows 7 Enterprise ,Windows 10 Enterprise
Microsoft Windows Server 2016 Datacenter,Server Datacenter (full installation)
Microsoft Windows 10 Business,Windows 10 Pro
Microsoft Windows Server 2012 R2 Datacenter,Server Datacenter (full installation)
Microsoft Windows Server 2016 Datacenter Evaluation,Server Datacenter (evaluation installation)
Microsoft Windows Server 2016 Standard,Server Standard
Microsoft Windows 10 Enterprise Evaluation,Windows 10 Enterprise Evaluation
Microsoft Windows 10 Pro N,Windows 10 Pro N
Microsoft Windows Server Datacenter,
Майкрософт Windows 10 Pro,Windows 10 Pro
Microsoft Windows 10 Home Single Language,Windows 10 Home Single Language
Microsoft Windows 10 Enterprise,Windows 10 Enterprise
Microsoft Windows 8.1 Pro,Windows 10 Pro
Microsoft Windows 10 Pro Education,
Microsoft Windows 10 Famille,Windows 10 Home
Microsoft Windows Server 2012 R2 Standard,Server Standard
Microsoft Windows Server 2012 R2 Foundation,Server Foundation
Microsoft Windows Server 2008 R2 Standard ,Server Standard
Microsoft Windows 10 Enterprise 2016 LTSB,
Microsoft Windows Server 2019 Standard,Server Standard
Microsoft Windows 10 Pro,Windows 10 Pro
Microsoft Windows Server 2016 Standard Evaluation,Server Standard (evaluation installation)
Microsoft Windows 7 Professional ,Windows 10 Pro
Майкрософт Windows 10 Домашняя для одного языка,Windows 10 Home Single Language
```

**Comment by muffins** [comment]
@directionless good call, I'm happy to hear what @FritzX6 has to say. I'm going to close this out, but am happy to do follow-up PRs if folks think it's necessary. Overall I feel this might be worth just calling `codename` whatever `name` is. I don't think that `codename` gives that much extra value for the cost of maintaining that list :\

**Comment by directionless** [comment]
Works for me!

**Pull request merged**